CMakeLists.txt

```diff
@@ -344,6 +345,15 @@ if (NLOPT_GUILE)
   find_package (Guile)
 endif ()
 
+if (NLOPT_JAVA)
+  # we do not really need any component, only the main JNI target, but if the
+  # list of components is left empty, FindJNI defaults to "JVM AWT", and we
+  # specifically do not want to check for the AWT library that is not available
+  # on headless installations
+  find_package (JNI COMPONENTS JVM)
+  find_package (Java)
```

Comment:
we could specify the minimum version here instead of test/CMakeLists.txt

Reply:
But as the comment and the commit message both clearly explain, the bindings will build with basically any version, probably as old as 1.4, maybe even 1.0. Only the test program has 3 lines that require 1.8. So it is wrong to require 1.8 here.

If you insist that the test needs to support everything the binding supports, I can change those to use explicit anonymous classes, but the result is ugly and I worry that users will copy&paste it even if they do not actually need support for ancient Java versions. (I expect users to use the test class as an example.)

Reply:
looks like java 1.8 is pretty old (2014), maybe its a reasonable requirement